File: hashen_bitproject_1800135_new/server/src/main/java/bit/project/server/entity/Reconnectionrequest.java
package bit.project.server.entity;

import lombok.Data;
import java.util.List;
import javax.persistence.*;
import java.time.LocalDate;
import javax.persistence.Id;
import javax.persistence.Lob;
import java.time.LocalDateTime;
import lombok.NoArgsConstructor;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

@Data
@Entity
@NoArgsConstructor
@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class Reconnectionrequest{
    @Id
    @GeneratedValue(strategy=GenerationType.IDENTITY)
    private Integer id;

    private String code;

    private LocalDate date;

    @Lob
    private String description;

    private LocalDateTime tocreation;


    @ManyToOne
    private Connection connection;

    @ManyToOne
    private Reconnectionrequeststatus reconnectionrequeststatus;

    @ManyToOne
    @JsonIgnoreProperties({"creator","status","tocreation","roleList"})
    private User creator;


    @JsonIgnore
    @OneToMany(mappedBy = "reconnectionrequest")
    private List<Taskallocation> reconnectionrequestTaskallocationList;


    public Reconnectionrequest(Integer id){
        this.id = id;
    }

    public Reconnectionrequest(Integer id, String code, Connection connection){
        this.id = id;
        this.code = code;
        this.connection = connection;
    }

}

File: hashen_bitproject_1800135_new/client/src/app/entities/reconnectionrequest.ts
import {User} from './user';
import {Connection} from './connection';
import {DataPage} from '../shared/data-page';
import {Reconnectionrequeststatus} from './reconnectionrequeststatus';

export class Reconnectionrequest {
  id: number;
  code: string;
  connection: Connection;
  date: string;
  reconnectionrequeststatus: Reconnectionrequeststatus;
  description: string;
  creator: User;
  tocreation: string;
}

export class ReconnectionrequestDataPage extends DataPage{
    content: Reconnectionrequest[];
}
